File: basic-problem-sets/CountLetters.py
def count_letters(string):
    letter_count = {}
    for char in string:
        if char not in letter_count:
            letter_count[char] = 1
        else:
            letter_count[char] += 1
            
    return letter_count

user_input = input()
print(count_letters(user_input))

File: basic-problem-sets/CountLetters.cpp
#include <iostream>
#include <string>
#include <map>

std::map<char, int> LettersCount(const std::string& str) {
    std::map<char, int> letterCount;
    for (char ch : str) {
        ++letterCount[ch]; }
    return letterCount; }

int main() {
    std::string input;
    
    std::getline(std::cin, input);

    std::map<char, int> output = LettersCount(input);

    for (const auto& pair : output) {
        std::cout << pair.first << ": " << pair.second << std::endl; } }
